### ISSUE TITLE
Could you provide some hyper-parameters and results?

### ISSUE THREAD

**Issue by xuguodong03** [dwang181/active-mixup#1]
Hi, I am trying to reimplement this paper recently. And I have some questions about the training hyper-parameters. 

Take CIFAR10 as example, you said that you followed the training setting of ZSKD. In ZSKD, they set learning rate 0.001, batch size 512, epoch 1000.  Do the results in your paper (teacher83.07 and mixup 74.60) come from this hyper-parameters? If not, could you tell me your hyper-parameters?

Also, I notice that in ZSKD Table 3, the KD student's accuracy surpass the CE student by a really small margin (0.04%). This is a strange result. In general, the KD's gain will not be so small. In your paper, you didn't mention the result of CE student. So I wonder could you tell me the results. Thanks a lot!

**Comment by dwang181** [comment]
Hello there, Thanks very much for these questions. 
1. Our teacher model (AlexNet) with 83.07% top-1 accuracy is trained with: batch size=128  lr=0.01, SGD with 0.9 momentum, 80-60-40-20 epochs with 0.1 lr rate decay, and weight_decay=5.0e-4. Mixup process is trained with: batch size=128 lr=0.1, SGD with 0.9 momentum, 80-60-40-20 epochs with 0.1 rate decay, and weight_decay=5.0e-4. Here, the training/distillation for AlexNet may show some strange intermediate results. If you run into related problems, just let me know.
2. For ZSKD, we adopted the results reported in the paper directly and did not reproduce their results. I would recommend that you may contact the authors of that paper for specific questions and detailed answers.

**Comment by xuguodong03** [comment]
Thanks for your reply. I need another two results to verify if I implement the paper correctly. I will be very grateful if you could provide them.
The first one is the accuracy (in cifar10) of naive student (with cross-entropy loss on the complete dataset) in your training setting, namely batch size=128, lr=0.01....
The second one is the accuracy of active mixup student in step0 (trained with initial 2000 unlabeled data). I want to know the gain brought by active mixup. 
Thanks a lot!

**Comment by dwang181** [comment]
Hello there, 

For Table 1 with AlexNet training,
The first one: we did not train student model with all labeled data. I may have a try. My hunch told me it should be close to ~83% in our setting, but I will do double-check.
The second one: ~35%. It has some variance and depends on initial 2000 images. 

For Table 2 with VGG16 training,
The first one: Same as above. It may be close to ~93%, but I will do double-check.
The second one: ~68.09% which is located within the first row. It may also show some variance, but should not be much.

If you just want to check if the implementation is correct, I suggest you check the results with Table 2 (VGG16) first. If you find any outrageous results, just let me know.

**Comment by xuguodong03** [comment]
Thanks, I will theck the Table 2. Is the student of this experiments VGG16-half (reduce all layers' channels by half)?

Another question is how do you solve quadratic expansion brought by mixup. For instance, you have 16k unlabel data. If each image can mixed with any other image to generate 11 (lam from 0.3 to 0.7) mixup images, then you will get 16000x8000x11 = 1.4 billion images. If there any special strategy to solve this issue?

**Comment by dwang181** [comment]
Hello there, 

Q1: Is the student of this experiments VGG16-half (reduce all layers' channels by half)?
No. The student network is VGG-16. Our problem setting is to distill teacher model for model reproduction instead of model compression. Also, blackbox here implies no knowledge of model architecture/size. Therefore, it is not necessary to reduce network size. Just in case you are confused. 

Q2: Another question is how do you solve quadratic expansion brought by mixup. For instance, you have 16k unlabel data. If each image can mixed with any other image to generate 11 (lam from 0.3 to 0.7) mixup images, then you will get 16000x8000x11 = 1.4 billion images. If there any special strategy to solve this issue?
Good question. It is necessary to obtain this large image pool. It is inevitable for our current work. For processing, it could be optimized by some engineering work, like better parallelization; however, multiple GPUs are critical, in my opinion.
